### DISCUSSION TITLE
error handling on material ui select

### DISCUSSION THREAD

**Discussion by DonnaTelloo** [react-hook-form/react-hook-form#4969] [Q&A]
Hi, I am using React Hook Form@5.x.x
I have reusable select component, where I pass control and name for Controller.
My code:
`<FormControl
      variant="filled"
      error={Boolean(error)}
      style={{
        width: "100%",
        marginBottom: 25,
      }}>
      {control !== undefined ? (
        <>
          <Controller
            name={name}
            control={control}
            rules={{ required: true }}
            as={
              <>
                <InputLabel id="demo-simple-select-filled-label">
                  {label}
                </InputLabel>
                <Select
                  labelId="demo-simple-select-filled-label"
                  className={classes.root}
                  // inputRef={selectRef}
                  // onChange={(event) => event.target.value}>
                >
                  {children}
                </Select>
              </>
            }></Controller>
          {error !== undefined && (
            <FormHelperText>{error.message && error.message}</FormHelperText>
          )}
              </FormControl>
`

but it does not take any value, also no error. 

I am using Yup validation schema and is it possible to pass validation from here? and not use rules on controller

**Comment by DonnaTelloo** [comment]
![image](https://user-images.githubusercontent.com/23365416/116135384-f0751c00-a685-11eb-80f8-cb66c4132a56.png)
![image](https://user-images.githubusercontent.com/23365416/116135444-02ef5580-a686-11eb-8710-41b545f522fc.png)
![image](https://user-images.githubusercontent.com/23365416/116135472-0aaefa00-a686-11eb-843e-dd7ef7072324.png)

**Reply by bluebill1049** [reply]
v5 you may need to change onChange name 

https://react-hook-form.com/v5/api#Controller

<img width="986" alt="Screen Shot 2021-04-27 at 10 44 51 am" src="https://user-images.githubusercontent.com/10513364/116168386-99cf1700-a745-11eb-958f-3ceaad29779c.png">
